### DISCUSSION TITLE
Grafana Plugin

### DISCUSSION THREAD

**Discussion by Martin2877** [tenzir/tenzir#3349] [Ideas & Features]
### Use Case

Currently there is not a web view to operate vast for search and preview data. Are there any plan? A grafana-vast plugins may meet this need.

### Feature Description

A tool with search panel that connect to vast .

**Comment by mavam** [comment]
Thanks for reaching out, @Martin2877.

Quick clarification question: would you like to render arbitrary data *in* VAST, or would you like to visualize VAST metrics?

**Comment by Martin2877** [comment]
> 
The former one，a front-end tool for exploring and working with data , like https://github.com/brimdata/zui

**Comment by mavam** [comment]
@Martin2877 thanks for clarifying. We're actually releasing something quite similar later this year! Check out tenzir.com to join the waitlist. If you're interested in early access as part of our beta testing, shoot me a mail at matthias@tenzir.com.

Regarding Grafana itself, it seems that [adding a data source](https://grafana.com/tutorials/build-a-data-source-plugin/) is a well-documented process. In fact, since Grafana also models data as [data frames](https://grafana.com/docs/grafana/latest/developers/plugins/data-frames/), there might be a very natural integration:

> The data frame structure is inspired by, and uses the [Apache Arrow Project](https://arrow.apache.org/). Javascript Data frames use Arrow Tables as the underlying structure, and the backend Go code serializes its Frames in Arrow Tables for transmission.

VAST uses Arrow natively, so in theory there should be a high-bandwidth access path to the data for Grafana. I haven't looked deeper yet, but it looks like the path is paved here.

**Reply by mavam** [reply]
This came across my feed in this context:
- https://sneller.ai/blog/one-billion-records-on-grafana/
- https://github.com/SnellerInc/grafana-datasource
